src/abstractdataframe/reshape.jl

```diff
@@ -397,9 +397,25 @@ function unstack(df::AbstractDataFrame, rowkeys, colkey::ColumnIndex,
                  values::ColumnIndex; renamecols::Function=identity,
                  allowmissing::Bool=false, allowduplicates::Bool=false,
                  valuestransform=nothing, fill=missing)
+    # first make sure that rowkeys are unique and
+    # normalize all selectors them as a strings
+    # if some of the selectors is wrong we will get an early error here
+    rowkeys = names(df, index(df)[rowkeys])
+    colkey = only(names(df, colkey))
+    values = only(names(df, values))
+
     if !isnothing(valuestransform)
-        groupcols = vcat(index(df)[rowkeys], index(df)[colkey])
-        @assert groupcols isa AbstractVector{Int}
+        # potentially colkey can be also part of rowkeys so we need to do unique
+        groupcols = unique([rowkeys; colkey])
```

Comment:
> Should we disallow this?

It is allowed now in DataFrames.jl 1.3 and I do not see a problem with it:
```
julia> unstack(DataFrame(x=["a", "b", "c"]), :x, :x, :x)
3×4 DataFrame
 Row │ x       a        b        c
     │ String  String?  String?  String?
─────┼───────────────────────────────────
   1 │ a       a        missing  missing
   2 │ b       missing  b        missing
   3 │ c       missing  missing  c
```
do you see any issues here? (the same column can be used for rows, columns and values if user thinks it is useful)

Reply:
No particular issue. I just find `unstack` already complex enough to reason about. But who knows what users may want to do... :-D

Reply:
Got it. Using `valuestransform=length` is indeed clever, and that's maybe the only case where you would want to reuse the column since you don't care about the particular column which is used.